source/extensions/tracers/opentelemetry/tracer.cc

```diff
@@ -245,6 +247,14 @@ void Tracer::flushSpans() {
 }
 
 void Tracer::sendSpan(::opentelemetry::proto::trace::v1::Span& span) {
+  if (span_buffer_.size() >= max_cache_size_) {
+    ENVOY_LOG(
+        warn,
+        "Span buffer size exceeded maximum limit. Discarding span. Current size: {}, Max size: {}",
+        span_buffer_.size(), max_cache_size_);
```

Comment:
This can log spam very badly. I would make this a backoff log variant or reduce the log level.

/wait